mm2src/mm2_db/src/indexed_db/db_lock.rs

```diff
@@ -1,78 +1,123 @@
 use super::{DbIdentifier, DbInstance, InitDbResult};
-use futures::lock::{MappedMutexGuard as AsyncMappedMutexGuard, Mutex as AsyncMutex, MutexGuard as AsyncMutexGuard};
 use mm2_core::{mm_ctx::MmArc, DbNamespaceId};
-use primitives::hash::H160;
+use std::collections::HashMap;
 use std::sync::{Arc, Weak};
+use tokio::sync::{Mutex as AsyncMutex, OwnedMappedMutexGuard, OwnedMutexGuard};
 
 /// The mapped mutex guard.
 /// This implements `Deref<Db>`.
-pub type DbLocked<'a, Db> = AsyncMappedMutexGuard<'a, Option<Db>, Db>;
+pub type DbLocked<Db> = OwnedMappedMutexGuard<Option<Db>, Db>;
 pub type SharedDb<Db> = Arc<ConstructibleDb<Db>>;
 pub type WeakDb<Db> = Weak<ConstructibleDb<Db>>;
 
+#[allow(clippy::type_complexity)]
 pub struct ConstructibleDb<Db> {
     /// It's better to use something like [`Constructible`], but it doesn't provide a method to get the inner value by the mutable reference.
-    mutex: AsyncMutex<Option<Db>>,
+    mutexes: Arc<AsyncMutex<HashMap<String, Arc<AsyncMutex<Option<Db>>>>>>,
```

Comment:
Could you clarify why are we wrapping list of mutexes with a parent mutex here?

Reply:
for thread safe mutability for list of initialized/non initialized indexeddb connections

Reply:
When you lock the parent mutex, there can't be data race in the inner values.

Reply:
Each connection entry in the HashMap wrapped in an AsyncMutex. This means that each database connection is individually locked and a thread accessing a specific database connection will lock only that connection and not the whole HashMap. Does that make sense ?

I think it's quite different from `mutexes: Arc<AsyncMutex<HashMap<String, Arc<Option<Db>>>>>,` where different thread will try to write the same time

Reply:
> Each connection entry in the HashMap wrapped in an AsyncMutex. This means that each database connection is individually locked and a thread accessing a specific database connection will lock only that connection and not the whole HashMap. Does that make sense ?

So initially the connection list is locked, then a particular connection gets locked, and then the initial lock gets released, right? If so, this is way too expensive (on each access to this connection list, there will be lots of context switches) and doesn't seem worth for what it's doing. Did you consider using [RwLock](https://doc.rust-lang.org/std/sync/struct.RwLock.html) lock for the connection list?

Reply:
Ok I substituted Mutex for RwLock and all working fine! thanks

Reply:
> Did you consider using [RwLock](https://doc.rust-lang.org/std/sync/struct.RwLock.html) lock for the connection list?

Just to clarify, this is only for connection list, not for the connections in the list. For connection entries, you should still use Mutex.